starlite/middleware/session/base.py

```diff
@@ -0,0 +1,313 @@
+    path: str = "/"
+    """Path fragment that must exist in the request url for the cookie to be valid. Defaults to '/'."""
+    domain: Optional[str] = None
+    """Domain for which the cookie is valid."""
+    secure: bool = False
+    """Https is required for the cookie."""
+    httponly: bool = True
+    """Forbids javascript to access the cookie via 'Document.cookie'."""
+    samesite: Literal["lax", "strict", "none"] = "lax"
+    """Controls whether or not a cookie is sent with cross-site requests. Defaults to 'lax'."""
+
+    @property
+    def middleware(self) -> DefineMiddleware:
+        """Use this property to insert the config into a middleware list on one
+        of the application layers.
+
+        Examples:
+
+            ```python
+            from os import urandom
+
+            from starlite import Starlite, Request, get
+            from starlite.middleware.session import SessionCookieConfig
+
+            session_config = SessionCookieConfig(secret=urandom(16))
+
+
+            @get("/")
+            def my_handler(request: Request) -> None:
+                ...
+
+
+            app = Starlite(route_handlers=[my_handler], middleware=[session_config.middleware])
+            ```
+
+        Returns:
+            An instance of DefineMiddleware including 'self' as the config kwarg value.
+        """
+        return DefineMiddleware(SessionMiddleware, backend=self._backend_class(config=self))
+
+
+class ServerSideSessionConfig(BaseBackendConfig):
+    session_id_bytes: int = 32
+    """
+    Number of bytes used to generate a random session-ID
+    """
+
+
+ConfigT = TypeVar("ConfigT", bound=BaseBackendConfig)
+ServerConfigT = TypeVar("ServerConfigT", bound=ServerSideSessionConfig)
+
+
+class SessionBackend(abc.ABC, Generic[ConfigT]):
+    def __init__(self, config: ConfigT) -> None:
+        """Generic session backend, for either client- or server-side
+        storage."""
+        self.config = config
+
+    @staticmethod
+    def serialise_data(data: "ScopeSession", scope: Optional["Scope"] = None) -> bytes:
+        """Serialise data into bytes for storage in the backend.
+
+        The serialiser will be
+        gathered from `scope` and fall back to [default_serializer][starlite.utils.default_serializer]
+        """
+        serializer = (get_serializer_from_scope(scope) if scope else None) or default_serializer
+        return dumps(data, default=serializer, option=OPT_SERIALIZE_NUMPY)
+
+    @staticmethod
+    def deserialise_data(data: Any) -> Dict[str, Any]:
+        """Deserialise data into a dictionary for use in the application
+        scope."""
+        return cast("Dict[str, Any]", loads(data))
+
+    @abc.abstractmethod
+    async def store_in_message(
+        self, scope_session: "ScopeSession", message: "Message", connection: ASGIConnection
+    ) -> None:
+        """Stores the necessary information in the outgoing `Message`
```

Comment:
```suggestion
        """Stores the session information in the outgoing `Message`
```

Reply:
Hmm. I feel like your suggestion would imply that the actual session data is stored in the `Message`, which *may* be the case (for the cookie session), but not always. My intention with expressing it this way was to convey that it stores the needed information, whatever this may be, so it *could* be the session data itself, but - e.g. in the case of the server backends - it could also just be a session-ID cookie.